base/applications/cmdutils/reg/import.c

```diff
@@ -772,11 +772,42 @@ static WCHAR *dword_data_state(struct parser *parser, WCHAR *pos)
 static WCHAR *hex_data_state(struct parser *parser, WCHAR *pos)
 {
     WCHAR *line = pos;
+#ifdef __REACTOS__
+    WCHAR Buffer[10] = { 0 };
+    WCHAR* ret;
+    BOOL unicode_in_ansi = FALSE;
+    BOOL result;
+#endif
 
     if (!*line)
         goto set_value;
 
+#ifdef __REACTOS__
+    if ((!parser->is_unicode) &&
+        (parser->data_type == REG_EXPAND_SZ) &&
+        (parser->parse_type == REG_BINARY))
+    {
+        memcpy(Buffer, pos, 18);
+        Buffer[9] = UNICODE_NULL;
+        ret = wcsstr(Buffer, L"00,"); // Any UNICODE characters?
+        unicode_in_ansi = (ret != NULL);
```

Comment:
I wonder why this kind of stuff is not done instead when the parser structure is initialized with data.

Reply:
I believe that it is because the parser just checks whether the input file is in ANSI or UNICODE. In this case, we have an ANSI file that is creating UNICODE data, so it is no quite so straightforward. Thanks.